packages/shared/src/util/features.ts

```diff
@@ -114,3 +116,54 @@ export function validateFeatureValue(
 
   return value;
 }
+
+type StaleFeatureReason =
+  | "error"
+  | "draft-state"
+  | "no-active-envs"
+  | "no-rules"
+  | "no-active-exps"
+  | "all-exps-onesided";
+
+export function isFeatureStale(
+  feature: FeatureInterface,
+  linkedExperiments: ExperimentInterface[] | undefined = []
+): { stale: boolean; reason?: StaleFeatureReason } {
+  if (feature.linkedExperiments?.length && !linkedExperiments.length) {
+    // eslint-disable-next-line no-console
+    console.error("isFeatureStale: linkedExperiments not provided");
+    return { stale: false, reason: "error" };
+  }
+
+  const twoWeeksAgo = subWeeks(new Date(), 2);
+  // TODO why is dateUpdated a string?
+  const stale = new Date(feature.dateUpdated) < twoWeeksAgo;
+
+  if (!stale) return { stale };
+
+  if (feature.draft) return { stale, reason: "draft-state" };
+
+  const envSettings = Object.values(feature.environmentSettings ?? {});
+
+  const noneEnabled = !envSettings.some((e) => e.enabled);
+  if (noneEnabled) return { stale, reason: "no-active-envs" };
```

Comment:
Instead of this being an explicit check, I think we should just filter the envSettings to only enabled environments.  The `rules.length === 0` check will then pick up this condition.

The current logic fails if `prod` is disabled and has rules, but `dev` is enabled and has no rules.

Reply:
Ah good catch